cfme/cloud/keypairs.py

```diff
@@ -0,0 +1,47 @@
+
+import cfme.fixtures.pytest_selenium as sel
+from cfme.common import SummaryMixin, Taggable
+from cfme.web_ui import Quadicon, flash, breadcrumbs, summary_title
+from functools import partial
+from cfme.web_ui import toolbar as tb
+from utils.wait import wait_for
+
+
+cfg_btn = partial(tb.select, "Configuration")
+
+
+class KeyPair(Taggable, SummaryMixin):
+    """ Model of KeyPairs
+
+    Args:
+        name: Name of Keypairs.
+    """
+
+    def __init__(self, name=None):
+        self.name = name
+
+    def _on_detail_page(self):
+        title = '{} (Summary)'.format(self.name)
+        return breadcrumbs() == ['Key Pairs', title] and summary_title() == title
+
+    def load_details(self, refresh=False):
+        if not self._on_detail_page():
+            self.navigate(detail=True)
+        elif refresh:
+            tb.refresh()
+
+    def delete(self):
+        sel.force_navigate('clouds_key_pairs', context={'keypairs': self})
+        cfg_btn("Remove this Keypairs from the VMDB", invokes_alert=True)
+        sel.handle_alert(cancel=False)
+        flash.assert_success_message('The selected Keypairs was deleted')
+
+    def nav_to_parameters_link(self):
+        sel.force_navigate('clouds_key_pairs', context={'keypairs': self})
+        self.summary.relationship.instances.click()
```

Comment:
Given the fact that summary navigates to the details page itself, you should be able to remove the force navigate.